File: slackoFrontend/slack/src/app/models/Voting.ts
import {Round} from './Round';
import {Book} from './Book';

export class Voting {
  firstRound: Round;
  secondRound: Round;
  thirdRound: Round;
  id: number;
  isActive: boolean;
  winner: Book;
  timeToNextVotingInSec: number;

  constructor(firstRound: Round, secondRound: Round, thirdRound: Round, id: number, isActive: boolean, winner: Book, timeToNextVotingInSec: number) {
    this.firstRound = firstRound;
    this.secondRound = secondRound;
    this.thirdRound = thirdRound;
    this.id = id;
    this.isActive = isActive;
    this.winner = winner;
    this.timeToNextVotingInSec = timeToNextVotingInSec;
  }
}

File: slackoBackend/src/main/java/com/slack/entities/Voting.java
package com.slack.entities;

import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import javax.persistence.*;
import java.io.Serializable;

@Entity
@Data
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Voting implements Serializable {

    @Id
    @Setter(AccessLevel.NONE)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REMOVE},
                fetch = FetchType.LAZY)
    @JoinColumn(name = "first_round")
    private Round firstRound;

    @OneToOne(cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY)
    @JoinColumn(name = "second_round")
    private Round secondRound;

    @OneToOne(cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY)
    @JoinColumn(name = "third_round")
    private Round thirdRound;

    @Transient
    private Boolean isActive;

    @OneToOne(cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY)
    @JoinColumn(name = "round_winner")
    private Book winner;

    @Transient
    private Integer timeToNextVotingInSec;
}
